src/StateService/Storage/StateSnapshot.cs

```diff
@@ -0,0 +1,45 @@
+using Neo.IO;
+using Neo.IO.Caching;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.Plugins.MPT;
+using System;
+
+namespace Neo.Plugins.StateService.Storage
+{
+    public class StateSnapshot : IDisposable
+    {
+        private readonly ISnapshot snapshot;
+        public MetaDataCache<HashIndexState> LocalRootHashIndex;
+        public MetaDataCache<HashIndexState> ValidatedHashIndex;
+        public DataCache<SerializableWrapper<uint>, StateRoot> StateRoots;
```

Comment:
Can you use `IStore` directly? Because I think `MetaDataCache` and `DataCache` are going to be removed soon. neo-project/neo#2215

Reply:
Done.